### ISSUE TITLE
Create 10-15 fine-tuning examples for your aspect

### ISSUE THREAD

**Issue by divilian** [divilian/frozone#114]
Create a `.csv` file in the `data/frobot_training` directory for your "bad dialogue aspect" (logical fallacies, misinformation, etc.) with this header row:
```
user_message,ai_response,dialogue_id,response_id,episode_done,issue_labels
```
See `toxicity.csv` for examples. The fields are:

* `user_message`: A sample user text, prepended with an identifying letter (like "`A: `").
* `ai_response`: Either the phrase "`(pass)`" or a substantive Frobot response, with _no_ prepended identifying letter.
* `dialogue_id`: A numeric identifier for a single dialogue back-and-forth chain, **unique within your `.csv` file**. (@nyfine99 will make these globally unique when he merges all these files into a single fine-tuning dataset).
* `response_id`: The number of the text/response (starting with 0) within the dialogue back-and-forth chain.
* `episode_done`: If this is the last text/response in the chain, `True`, otherwise, `False`. (The way @divilian did it, all rows marked `True` had substantive Frobot replies; all rows marked `False` were "`(pass)`". This is not strictly necessary, however.)
* `issue_labels`: A Python-list-demarcated list of labels, purely for organizational purposes. (@divilian's say "`[misrepresentation]'".)

Notes

1. Although it's not strictly necessary to put `user_message` and `ai_response` field entries in quotes, doing so is always safe, and is necessary if you have any quotes or commas in the message/response.
2. If you do have quotation marks inside your `user_message`/`ai_response` fields, you should **double**-double-quote them, like this:
```
"A: Yo, jerk, stop calling me ""MAGA""!","Okay, I'll stop.",0,0,True,['formatting example']
```
3. Before you declare victory, load your `.csv` file into Polars or Pandas to verify it's parsed correctly and all the fields line up. (Hint: they won't, at first; there are just too many opportunities for clerical errors.) One way to do this is to type this in Python:
```
import polars as pl
x = pl.read_csv('theNameOfYourCsv.csv',comment_prefix='#')
print(x[['dialogue_id',response_id','episode_done','issue_labels']]
```
and make sure it all looks legit.

**Comment by nyfine99** [comment]
> Although it's not strictly necessary to put user_message and ai_response field entries in quotes, doing so is always safe, and is necessary if you have any quotes or commas in the message/response.

This may be misleading and could lead to issues like the one I experienced during my debugging when creating my own training data. In an effort to avoid making the same mistakes:

- If editing in excel, you likely do not need to add quotes of any sort, except when the speaker is quoting something.
- I'm not sure that quotes are always safe, and if I recall correctly I had trouble with quotes when there was no comma. Please *do not* include quotes unless there is a comma in the message.

However, if you can get read_csv to work correctly, then whatever you made is fine!

**Comment by Garrett-Mckenzie** [comment]
You want abbreviations, bad grammar, and misspellings in both as that is what our deployment data will be most like. No need to overdo it but sprinkle it in for sure.

Garrett McKenzie

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by nyfine99** [comment]
Quick update: I have built the function to fuse these into one jsonl file. See data/training_data_to_jsonl.py
